File: backend/app/schemas/parent.py
from pydantic import BaseModel, Field
from typing import Any, Optional
from datetime import datetime


class ParentStudentLinkCreate(BaseModel):
    """Request to link parent to student."""
    student_email_or_username: str = Field(..., description="Email or username of the student to link with")


class ParentStudentLinkResponse(BaseModel):
    """Parent-student link response."""
    id: int
    parent_id: int
    student_id: int
    student_name: str
    student_email: str
    student_username: str
    status: str
    requested_at: datetime
    approved_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class ParentStudentLinkPending(BaseModel):
    """Pending link request for admin review."""
    id: int
    parent_name: str
    parent_email: str
    parent_username: str
    student_name: str
    student_email: str
    student_username: str
    requested_at: datetime

    class Config:
        from_attributes = True


class LinkApprovalRequest(BaseModel):
    """Admin approval/rejection of link."""
    action: str = Field(..., pattern="^(approve|reject)$")
    reason: Optional[str] = None


class StudentProgressSummary(BaseModel):
    """Student progress summary for parent view."""
    student_id: int
    student_name: str
    student_username: str
    total_attempts: int
    average_score: Optional[float] = None
    last_attempt_at: Optional[datetime] = None
    recent_attempts: list = Field(default_factory=list)

    class Config:
        from_attributes = True


class StudentDetailForParent(BaseModel):
    """Detailed student info for parent view."""
    student_id: int
    student_name: str
    student_username: str
    email: str
    total_attempts: int
    average_score: Optional[float] = None
    standards_attempted: int
    recent_attempts: list = Field(default_factory=list)

    class Config:
        from_attributes = True


class ParentAssistantChatRequest(BaseModel):
    """Parent assistant chat request with optional UI-selected context."""
    message: str = Field(..., min_length=1, max_length=1000)
    thread_id: Optional[int] = None
    student_id: Optional[int] = None
    subject_id: Optional[int] = None
    grade_id: Optional[int] = None


class ParentAssistantChatResponse(BaseModel):
    """Parent assistant response for learning, syllabus, and quiz assignment tools."""
    intent: str
    answer: str
    thread_id: Optional[int] = None
    tool_name: Optional[str] = None
    requires_student: bool = False
    requires_subject: bool = False
    suggestions: list[str] = Field(default_factory=list)
    data: dict[str, Any] = Field(default_factory=dict)

File: frontend/src/types/parent.ts
export interface ParentStudentLink {
  id: number
  parent_id: number
  student_id: number
  student_name: string
  student_email: string
  student_username: string
  status: 'pending' | 'approved' | 'rejected'
  requested_at: string
  approved_at: string | null
}

export interface StudentDetailForParent {
  student_id: number
  student_name: string
  student_username: string
  email: string
  total_attempts: number
  average_score: number | null
  standards_attempted: number
  recent_attempts: DetailedAttempt[]
}

export interface DetailedAttempt {
  answer_id: number
  question_id: number
  standard_code: string | null
  standard_description: string | null
  is_correct: boolean
  answered_at: string | null
}

export interface LinkRequestData {
  student_email_or_username: string
}

export interface LinkRequestResponse {
  message: string
  link_id: number
  status: string
}

export interface ParentAssistantChatRequest {
  message: string
  student_id?: number
  subject_id?: number
  grade_id?: number
}

export interface ParentAssistantChatResponse {
  intent: string
  answer: string
  requires_student: boolean
  requires_subject: boolean
  suggestions: string[]
  data: Record<string, unknown>
}
